### PR TITLE
Snippet template proposal

### LINKED ISSUE #4: [CHORE] Snippet Template
We don't have a template for snippets in this repository yet. We should figure this out and create a template as soon as possible.

### PR DESCRIPTION
**Pull request by Chalarangelo**
Resolves #4.

This is the snippet template that we have sort of decided on over the last couple of weeks. Please comment below so we can conclude on a format and start working on #1.

### CODE REVIEW

**Comment by skatcat31** [comment]
LGTM

**Comment by skatcat31** [comment]
The only question I still have is how do we enforce the constructor -> properties -> methods -> render layout? Are we going to be putting that in a separate place?

File: snippet-template.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ReactDOM.render(<ComponentName />, mountNode);
+```
+
+<!-- tags: (separate each by a comma) -->
+
+<!-- expertise: (0,1,2) -->
+<!-- Expertise levels:
+	0: beginner
+	1: intermediate
+	2: advanced
```
Comment (skatcat31) [inline snippet-template.md]: We will want an expert level here. Eventually there will be a use case, even if it's archived, for render properties.

**Comment by skatcat31** [comment]
LGTM

**Comment by skatcat31** [comment]
Now we just need a "how to read" section and a "ctrl+f" section and we're golden XD

**Comment by Chalarangelo** [comment]
@skatcat31 Soon...

**Comment by Robe21** [comment]
😁

**Comment by meetzaveri** [comment]
@Chalarangelo I think there are projects/products which are built using older versions(still using older versions like 15.6 or 16.0) like some would not support  **new Context API** or some would be using **deprecated lifecycles** hence making it legacy code.

So there should be line added in snippet.md stating which versions or above versions this snippet can be used

**Comment by Chalarangelo** [comment]
@meetzaveri That's a good point. We should include a line in the README that states which version we are supporting (probably the latest as of right now) like we do in 30code for example where we state that we support ES6 and that there might be a need for transpilation for older projects.

**Comment by meetzaveri** [comment]
Yes exactly! I was thinking to make it project specific or snippet specific. So you just cleared the point about that

**Pull request merged**